runtime/codert_vm/znathelp.m4

```diff
@@ -431,6 +431,25 @@ BEGIN_FUNC(jitRunOnJavaStack)
     BRANCH_VIA_VMTHREAD(J9TR_VMThread_tempSlot)
 END_CURRENT
 
+BEGIN_HELPER(jitReadBarrier)
```

Comment:
I've removed this `jitReadBarrier` and switched to `HANDLE_SOFTWARE_READ_BARRIER (a.k.a. handleSoftwareReadBarrier)` in zcinterp.m4.

Commit a1cbc63486 renames software read barrier assembly code so all platforms have the same name.  `jitReadBarrier` is renamed to `handleSoftwareReadBarrier` and `TR_readBarrier` is renamed to `TR_softwareReadBarrier`. 

Although the helpers on x/p/z are all named `handleSoftwareReadBarrier`, they are used differently. On platform z, `TR_softwareReadBarrier` is not defined in the JIT as a `TR_Helper` because `handleSoftwareReadBarrier` is already a VM routine. It's entry point is registered in `vmThread->gsParameters.handlerAddr`. Hence, invocation of `handleSoftwareReadBarrier` on z is different from that on x86 and PPC.